plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hms/TrinoHiveCatalog.java

```diff
@@ -256,6 +264,44 @@ public Transaction newCreateTableTransaction(
                 isUsingSystemSecurity ? Optional.empty() : Optional.of(session.getUser()));
     }
 
+    @Override
+    public void registerTable(ConnectorSession session, SchemaTableName schemaTableName, String metadataFileLocation)
+    {
+        TrinoFileSystem fileSystem = fileSystemFactory.create(session);
+
+        validateLocation(fileSystem, metadataFileLocation);
+        TableMetadata tableMetadata = readMetadata(fileSystem, metadataFileLocation);
+        Optional<String> owner = getOwner(session);
+
+        io.trino.plugin.hive.metastore.Table.Builder builder = io.trino.plugin.hive.metastore.Table.builder()
```

Comment:
Extract the common code  used to create a table in the metastore from `io.trino.plugin.iceberg.catalog.hms.AbstractMetastoreTableOperations#commitNewTable` and `io.trino.plugin.iceberg.catalog.hms.TrinoHiveCatalog#registerTable` to a utility method.

Reply:
Created the `HiveIcebergUtil` class and moved the table-building code to a utility method.

Reply:
The implementation for building a Table is now slightly different from `io.trino.plugin.iceberg.catalog.hms.AbstractMetastoreTableOperations#commitNewTable`. So inlined the building table logic here.